mysql-test/suite/binlog/t/binlog_rows_event_max_rows.test

```diff
@@ -0,0 +1,29 @@
+source include/have_binlog_format_row.inc;
+source include/master-slave.inc;
+
+# setup
+let $MYSQLD_DATADIR = `select @@datadir`;
+let $MYSQLD_SECURE_FILE_DIR = `select @@secure_file_priv`;
+set @@global.binlog_rows_event_max_rows= 1;
+create table t1 (a int);
+
+# each insert should be in a separate row event because binlog_rows_event_max_rows = 1
+insert into t1 values(1), (2), (3);
+exec $MYSQL_BINLOG -v -v $MYSQLD_DATADIR/master-bin.0* | grep "Write_rows" | wc -l;
+
+# each update should be in a separate row event because binlog_rows_event_max_rows = 1
+update t1 set a = 10;
+exec $MYSQL_BINLOG -v -v $MYSQLD_DATADIR/master-bin.0* | grep "Update_rows" | wc -l;
+
+# there should be two delete events, 1st one with 2 rows and 2nd one with one
+# row because binlog_rows_event_max_rows = 2
+set @@global.binlog_rows_event_max_rows= 2;
+delete from t1;
+exec $MYSQL_BINLOG -v -v $MYSQLD_DATADIR/master-bin.0* | grep "Delete_rows" | wc -l;
+
+# cleanup
+drop table t1;
+set @@global.binlog_rows_event_max_rows= default;
```

Comment:
Save and restore the value of `binlog_rows_event_max_rows` rather than rely on `DEFAULT`